### PR TITLE
{lib}[foss/2021b] AMGX v2.3.0

### PR DESCRIPTION
**Pull request by lcniel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera41-1 - Linux Rocky Linux 8.7, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, 1 x NVIDIA Tesla T4, 525.85.12, Python 3.6.8
See https://gist.github.com/lcniel/4bd3bd6f9d0da050643a0307659ad77d for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-r07-01 - Linux Rocky Linux 8.7, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 1 x NVIDIA NVIDIA A40, 525.85.12, Python 3.6.8
See https://gist.github.com/lcniel/2dd40f83f3e2ff1e2cbca67c8d104152 for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
alvis3-28 - Linux Rocky Linux 8.7, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 525.85.12, Python 3.6.8
See https://gist.github.com/lcniel/56ffaa638c70cffd416e022218891398 for a full test report.

File: easybuild/easyconfigs/a/AMGX/AMGX-2.3.0-foss-2021b-CUDA-11.4.1.eb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ('CMake', '3.21.1'),
+]
+
+dependencies = [
+    ('CUDA', '11.4.1', '', SYSTEM),
+    ('UCX-CUDA', '1.11.2', versionsuffix),
+    ('magma', '2.6.2', versionsuffix),
+]
+
+#configopts = ' -DCUDA_NVCC_FLAGS="-fopenmp" '
```
Comment (boegel) [inline easybuild/easyconfigs/a/AMGX/AMGX-2.3.0-foss-2021b-CUDA-11.4.1.eb:29]: This looks like it should be cleaned up?

File: easybuild/easyconfigs/a/AMGX/AMGX-2.3.0-foss-2021b-CUDA-11.4.1.eb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+homepage = 'https://github.com/NVIDIA/AMGX'
+
+description = """Distributed multigrid linear solver library on GPU"""
+toolchain = {'name': 'foss', 'version': '2021b'}
+toolchainopts = {'openmp': True}
+
+github_account = 'NVIDIA'
+source_urls = [GITHUB_SOURCE]
+sources = ['v%(version)s.tar.gz']
+#patches = ['AMGX-2.3.0_fix_tests.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/a/AMGX/AMGX-2.3.0-foss-2021b-CUDA-11.4.1.eb:16]: no patch needed, so remove commented out line?

File: easybuild/easyconfigs/a/AMGX/AMGX-2.3.0-foss-2021b-CUDA-11.4.1.eb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+local_tests += 'CsrSparsityTests_Poisson7_100_100 CsrSparsityTests_Poisson7_10_10 CsrSparsityTests_Poisson9_100_100 '
+local_tests += 'CsrSparsityTests_Poisson9_10_10 DenseLUSolverTest_Factorization_Id_256 '
+local_tests += 'DenseLUSolverTest_Factorization_Id_32 DenseLUSolverTest_Solve_Id_256 DenseLUSolverTest_Solve_Id_32 '
+local_tests += 'DenseLUSolverTest_Solve_Poisson3D FGMRESConvergencePoisson FactoriesTest '
+local_tests += 'GenericSpMVTest IDRConvergencePoisson '
+local_tests += 'IDRmsyncConvergencePoisson LargeMatricesSupport MatrixTests MatrixVectorMultiplyTests NestedSolvers '
+local_tests += 'NormTests PermuteTests RandomMatrix SmootherBlockPoissonTest '
+local_tests += 'TemplateConfigTest TemplateTest VectorTests truncateCountTest '
+
+# runtest = "amgx_tests_launcher && tests/amgx_tests_launcher %s " % local_tests
```
Comment (boegel) [inline easybuild/easyconfigs/a/AMGX/AMGX-2.3.0-foss-2021b-CUDA-11.4.1.eb:52]: Why disable the tests by default?

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by lcniel** [comment]
Closing since I don't think I am likely to get back to this one soon, the test stuff needs to be worked out and checked with newer versions and right now I don't have time. Might as well do it on a later version.

**Pull request closed without merging**